### DISCUSSION TITLE
webdav连接失败，报405 method not allowded,求帮助

### DISCUSSION THREAD

**Discussion by beierweiwei** [AlistGo/alist#1751] [Q&A]
如题，网上找不到相关问题，有没有大佬给指指路

**Comment by 849506054** [comment]


**Reply by jihanc** [reply]
如果你是使用的浏览器访问的话，alist的webdav没有前端页面。

**Reply by 849506054** [reply]
好的，感谢解答

**Reply by githubgj** [reply]
我挂载的是移动139网盘，用rclone上传文件，同样是小文件可以上传，大文件上传没多久就报405 method not allowded，请问大家后来解决了吗？

**Reply by gfgjs** [reply]
> 如果你是使用的浏览器访问的话，alist的webdav没有前端页面。

一语惊醒梦中人😂我说怎么弄都"Method Not Allowed"

**Reply by coolmyp** [reply]
路径写：/dav

**Reply by mavericklou** [reply]
路径原来写dav可以，现在必须写/dav。这改动不说明清楚，坑死人啊

**Reply by remixjc** [reply]
请问大佬们，你们说的dav路径是啥玩意啊，不是链接后面加个就行吗

**Reply by oldsento** [reply]
> 路径原来写dav可以，现在必须写/dav。这改动不说明清楚，坑死人啊

请问在哪里改

**Comment by yymyyy** [comment]
碰到了。换用admin用户暂时正常。

**Reply by githubgj** [reply]
![issue](https://user-images.githubusercontent.com/102462164/208556028-de823378-8d61-43da-95cb-15c9a02be75c.png)

**Reply by githubgj** [reply]
我在VPS这里用admin账号登录rclone操作，还是提示405错误

**Comment by hiahia250** [comment]
相同情况

**Reply by Jiayou-Chao** [reply]
你是在Windows上操作的吗， 好像Windows的webdav协议不支持大文件，我改到Linux上使用以后就好了

**Reply by hiahia250** [reply]
在ubuntu上，rclone通过webdav挂载的alist，用rclone命令行几乎复制所有文件都报405 method not allowded

**Reply by hiahia250** [reply]
从onedrive复制到alist挂载的阿里云盘和pikpak都报错，都是影音文件，从2-20G不等，现在是刮削的nfo文件（几kb大小)也报错。不知道这是alist,rclone还是网盘本身限制造成的

**Reply by hiahia250** [reply]
复制到123云盘也同样报错

**Reply by hiahia250** [reply]
rclone sync nice:media/dv alist:123180/nicedv -P
2023-05-24 00:18:07 ERROR : jeen-yuhs: A Kanye Trilogy (2022)/jeen-yuhs.坎耶.维斯特三部曲.Jeen-yuhs.A.Kanye.Trilogy.S01E02.1080p.H265-官方中字.mp4: Failed to copy: Method Not Allowed: 405 Method Not Allowed
2023-05-24 00:18:07 ERROR : jeen-yuhs: A Kanye Trilogy (2022)/jeen-yuhs.坎耶.维斯特三部曲.Jeen-yuhs.A.Kanye.Trilogy.S01E03.1080p.H265（中字）.mp4: Failed to copy: Method Not Allowed: 405 Method Not Allowed
2023-05-24 00:18:07 ERROR : jeen-yuhs: A Kanye Trilogy (2022)/jeen-yuhs.坎耶.维斯特三部曲.Jeen-yuhs.A.Kanye.Trilogy.S01E01.1080p.H265-官方中字.mp4: Failed to copy: Method Not Allowed: 405 Method Not Allowed

**Reply by hiahia250** [reply]
123盘我看了下，好像只有123能秒传的文件能复制过去，其它的报错

**Comment by rongchuan17** [comment]
相同情况 ,用obsidian挂在webdav时挂在失败（鉴权类型为basic），手机用es文件浏览器却可以正常挂在

**Reply by HuntercodeT** [reply]
不要用admin账户，重新创一个账户就好了

**Reply by CNYoki** [reply]
可以试试在你的webdav url后加上你的目录，例如`/home`。

**Comment by ititchuan** [comment]
我换了用户，无论是用admin还是其他的，都不行。线程数而也已经配置成了一个，都基本上尝试遍了，就是找不到解决方案

**Reply by coolmyp** [reply]
路径写：/dav

**Comment by mowang111** [comment]
如果是用docker容器创建的alist的话，主要是容器没有权限修改那个文件夹，可以尝试，将下面的uid和gid都改成0，对应root用户，重启后即可
![image](https://github.com/user-attachments/assets/8885e684-c93f-47fd-b16e-2b6ed53e33dd)

**Comment by mavericklou** [comment]
都不行啊，有人解决么？每次升级都出现致命问题，这可不行啊

**Reply by mavericklou** [reply]
路径需要从dav改成/dav

**Reply by coolmyp** [reply]
嗯，搞定就好。我前几天特意回复了下，提醒一下😂

**Comment by defeatd** [comment]
我也遇到复制大文件报405错误，成功解决了，给后来者一点参考
## 我的使用方式
1. 用官方最新一键脚本安装alist
2. alist挂载115，rclone挂载pikpak官方webdav和alist的webdav
3. 用rclone copy复制pikpak到alist 115，遇到405问题
报错了405
## 解决方式
1. alist新建了一个名为'rclone'的用户，给他webdav权限
2. 重启alist
3. rclone挂载alist的webdav，用'rclone'这个用户登录
然后用rclone copy复制大文件就没问题了

**Comment by zht369** [comment]
打开ａｌｉｓｔ．点击“用户”。把灰色的权限都打勾。然后保存即可。

**Reply by Z46I4W5476SJA** [reply]
我是傻子。。。我以为是我弄得nginx反向代理有问题，然后各种改配置文件，最后实在不行，我还升级了nginx版本到官方的最新版本 ，然后最新版本的配置文件放的路径都变了，然后配置文件也要求conf尾缀，最后还是解决不了，想起来github看看，看到了这条，发现我webdav没有勾选管理权限。。。。真是充实的一下午

**Reply by woshiekao** [reply]
为什么我的用户里面没有相关的权限设置

**Comment by shiquda** [comment]
我的解决方式：在Nginx的 `location /` 配置中添加下面几行：

```nginx
        proxy_set_header   Upgrade $http_upgrade;
        proxy_set_header   Connection "upgrade";
        proxy_pass_request_headers on;
```

**Comment by oldsento** [comment]
openlist的情况，浏览器打开 http://localhost:5244/dav 窝是直接改成 http://localhost:5244//dav就能访问了，但部分软件如aimp不行，真是头大
